File: app/src/main/java/com/cn29/aac/ui/location/LocationActivityModule.java
package com.cn29.aac.ui.location;

import android.arch.lifecycle.ViewModelProviders;
import com.cn29.aac.ui.location.vm.LastLocationViewModel;
import com.cn29.aac.ui.location.vm.LastLocationViewModelFactory;
import dagger.Module;
import dagger.Provides;

/**
 * Created by charlesng0209 on 2/10/2017.
 */

@Module
public class LocationActivityModule {

  @Provides
  LastLocationViewModel provideLastLocationVM(LastLocationViewModelFactory factory,
      LocationActivity activity) {
    return ViewModelProviders.of(activity, factory).get(LastLocationViewModel.class);
  }
}

File: app/src/main/java/com/cn29/aac/ui/location/LocationActivityModule.kt
package com.cn29.aac.ui.location

import androidx.lifecycle.ViewModelProvider
import com.cn29.aac.ui.location.vm.LastLocationViewModel
import com.cn29.aac.ui.location.vm.LastLocationViewModelFactory
import dagger.Module
import dagger.Provides

/**
 * Created by charlesng0209 on 2/10/2017.
 */
@Module
class LocationActivityModule {
    @Provides
    fun provideLastLocationVM(factory: LastLocationViewModelFactory,
                              activity: LocationActivity): LastLocationViewModel {
        return ViewModelProvider(activity, factory)
                .get(
                        LastLocationViewModel::class.java)
    }
}